bin/rpi-security.py

```diff
@@ -190,34 +190,16 @@ def telegram_send_file(file_path):
         logger.info('Telegram file sent: %s' % file_path)
         return True
 
-def arp_ping_macs(mac_addresses, repeat=1):
+def arp_ping_macs(mac_addresses):
     """
-    Performs an ARP scan of a destination MAC address to try and determine if they are present on the network.
+    Performs an ARP scan of a destination network to try and determine if monitored devices are present on the network.
     """
-    def _arp_ping(mac_address, ip_address):
-        result = False
-        answered,unanswered = srp(Ether(dst=mac_address)/ARP(pdst=ip_address), timeout=1, verbose=False)
-        if len(answered) > 0:
-            for reply in answered:
-                if reply[1].hwsrc == mac_address:
-                    if type(result) is not list:
-                        result = []
-                    result.append(str(reply[1].psrc))
-                    result = ', '.join(result)
-        return result
-    while repeat > 0:
-        if time.time() - alarm_state['last_packet'] < 30:
-            break
-        for mac_address in mac_addresses:
-            result = _arp_ping(mac_address, config['network_address'])
-            if result:
-                logger.debug('MAC %s responded to ARP ping with address %s' % (mac_address, result))
-                break
-            else:
-                logger.debug('MAC %s did not respond to ARP ping' % mac_address)
-        if repeat > 1:
-            time.sleep(2)
-        repeat -= 1
+    logger.info("thread running")    
+    while True:
+        if time.time() - alarm_state['last_packet'] > 300:
+            logger.debug('Attempting ARP scan of subnet to wake up monitored devices')
+            answered,unanswered = srp(Ether(dst='ff:ff:ff:ff:ff:ff')/ARP(pdst=config['network_address']), timeout=1, verbose=False)
```

Comment:
Interesting.  It's been almost 100% reliable in my setup.  It has false alarmed when we are home maybe three times in the month it's been running and those all were when my wife has first arrived home.  At least one of those she had wifi switched off on her phone.  I've never seen it arm when one of us is home.  Only once in a while have I noticed in the log that the ping needs to run more than once to wake up the phones.

I have many more problems with false alarms when we're not home.  I usually get 5 - 10 per day.  I've tried adjusting the sensitivity and even swapped out PIR modules to no avail.

As far as differences go, the biggest one is that both my monitored phones are android.  They have the wifi set to never sleep.

The usb wifi adapter in the RPi differs from yours: https://www.amazon.com/TP-Link-Wireless-External-Supports-TL-WN722N/dp/B002SZEOLG

Your original code worked well for me too, I changed it mainly to learn a bit about python.  Plus I had it in my mind that I wanted the notification straight away rather than to try to wake up the phones one last time.  That's probably because my phones seem to wake up easily!

Anyway, thanks or taking a look at this.  It's interesting that it behaves so differently.  I'd wager it is and android vs ios thing.